Yona/Yona/Challenges/TimeBucketChallenges.swift

```diff
@@ -203,6 +198,11 @@ class TimeBucketChallenges: UIViewController,UIScrollViewDelegate {
                     Loader.Hide(self)
                 }
                 self.activityCategoriesArray = activities!
+                if self.activityCategoriesArray.count > 0 {
+                    dispatch_async(dispatch_get_main_queue()) {
+                        self.addNewGoalButton.hidden = false
+                    }
```

Comment:
@chandanvarma009  self.addNewGoalButton.hidden = self.activityCategoriesArray.count > 0

use this instead above.